doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -392,8 +465,64 @@ stands for :n:`let @ident := fun {+ @binder} => @term in @term’`.
 
 .. index:: match ... with ...
 
-Definition by case analysis
----------------------------
+Definition by cases: match
+--------------------------
+
+.. insertgram term_match pattern_list_or2
+
+.. productionlist:: coq
+   term_match : match `case_item_list_comma` `return_type_opt` with `or_opt` `eqn_list_or_opt` end
+   case_item_list_comma : `case_item_list_comma` , `case_item`
+                        : `case_item`
+   return_type_opt : `return_type`
+                   : `empty`
+   as_return_type : `as_name_opt` `return_type`
+                  : `empty`
+   return_type : return `term100`
+   or_opt : |
+          : `empty`
+   case_item : `term100` `as_name_opt` `in_opt`
+   as_name_opt : as `name`
+               : `empty`
+   in_opt : in `pattern`
+          : `empty`
+   eqn_list_or_opt : `eqn_list_or`
+                   : `empty`
+   eqn_list_or : `eqn_list_or` | `eqn`
+               : `eqn`
+   eqn : `pattern_list_or` => `term`
+   pattern_list_or : `pattern_list_or` | `pattern_list_comma`
+                   : `pattern_list_comma`
+   pattern_list_comma : `pattern_list_comma` , `pattern`
+                      : `pattern`
+   pattern : `pattern10` : `term`
+           : `pattern10`
+   ident_opt : `ident`
+             : `empty`
```

Comment:
This definition is only used in the fix / cofix section, so it should be moved there.